app/utils/connections.server.ts

```diff
@@ -1,21 +1,9 @@
-import { createCookieSessionStorage } from 'react-router'
+// import { createCookieSessionStorage } from 'react-router'
 import { type ProviderName } from './connections.tsx'
 import { GitHubProvider } from './providers/github.server.ts'
 import { type AuthProvider } from './providers/provider.ts'
 import { type Timings } from './timing.server.ts'
 
-export const connectionSessionStorage = createCookieSessionStorage({
-	cookie: {
-		name: 'en_connection',
-		sameSite: 'lax', // CSRF protection is advised if changing to 'none'
-		path: '/',
-		httpOnly: true,
-		maxAge: 60 * 10, // 10 minutes
-		secrets: process.env.SESSION_SECRET.split(','),
-		secure: process.env.NODE_ENV === 'production',
-	},
-})
-
```

Comment:
I'm confused why changing this was necessary because it looks like we've now lost the ability to display errors as the user goes between different pages in the OAuth flow. And I don't think that this change was necessary for upgrading to Remix auth. Can you help me understand why this change was made?

Reply:
This is no needed anymore since Remix Auth doesn't receive a session storage instance anymore, instead the strategy uses [`@mjackson/headers`](https://www.npmjs.com/@mjackson/headers) to save the data it needs in a cookie during the auth process.

Then after the user is back in the app from GitHub it you need to save the data in your session storage yourself, AFAIK you already did this anyway so there's no change in that part.

Reply:
Also related to error handling, now the strategies always throw an error, so you handle them with a try/catch

Reply:
Great to know! Thanks!